odoo/addons/base/views/res_lang_views.xml

```diff
@@ -68,6 +68,15 @@
                                 <field name="time_format"/>
                                 <field name="week_start"/>
                             </group>
+                            <div class='mt-4 border border-3 p-3 text-muted col-lg-4'>
+                                <p class="text-center">Preview</p>
+                                <group>
+                                    <field name="date_format_preview" widget="lang_date" attrs="{'invisible': [('date_format', 'in', ['', False])]}"/>
```

Comment:
Would it be possible to implement it the same way as the other ones, using a python computed field (Char)?

Having a new JS widget seems very heavy.

Reply:
Hello @awa-odoo 

The reason is we want to show the translation based on the current record language not from the user language.
and we don't want to store all the translations for the formats
that can be anything, hence it is suitable to do in JS. so, the user can use any format and the DateTime formater will handle it automatically.

Thanks

Reply:
Hello @nla-odoo 

Still not convinced on this one.
You could use the methods in tools to format_date and format_time based on a specific given format:
https://github.com/odoo/odoo/blob/master/odoo/tools/misc.py#L1359

So you would depend on the date_format and time_format fields in a compute method and change your preview accordingly.
Maybe I'm missing your point though, I don't really understand the part "and we don't want to store all the translations for the formats".

From what I understand, we need to use a format that would be supported by the backend anyway, as this is going to be used in the backend too once the res.lang is configured. 

Thank you for investigating & cheers!